### ISSUE TITLE
Select component `selected` prop is not working correctly

### ISSUE THREAD

**Issue by woothu** [carbon-design-system/carbon-components-svelte#570]
In the `Select` component, when binding variable with selected

```
let selectedValue;

...
<Select
  ...
  bind:selected={selectedValue}
/>
```

Even if the `selectedValue` is undefined the first `SelectItem` is shown as selected, if `selectedValue` dynamically changes user also sees the first item selected but in fact it's not.

Additionally, the Select should have the `null` state with no value selected and an additional placeholder prop.

**Comment by metonym** [comment]
Have you tried initializing the selected value with a disabled/hidden SelectItem?

```svelte
<script>
  import {
    Select,
    SelectItem,
    SelectItemGroup,
  } from "carbon-components-svelte";
</script>

<Select labelText="Carbon theme" selected="0">
  <SelectItem value="0" text="Select a theme" disabled hidden />
  <SelectItemGroup label="Light theme">
    <SelectItem value="white" text="White" />
    <SelectItem value="g10" text="Gray 10" />
  </SelectItemGroup>
  <SelectItemGroup label="Dark theme">
    <SelectItem value="g90" text="Gray 90" />
    <SelectItem value="g100" text="Gray 100" />
  </SelectItemGroup>
</Select>
```

**Comment by alkanna** [comment]
Not working for me either @metonym , I tried the following and the first item is always selected no matter the value I pass into it :
```html
<Select
              light
              labelText="Mutuelle"
              bind:selected={selectedHealthInsurance}
            >
              {#each $insurances as insurance}
                <SelectItem value={insurance.id} text={insurance.name} />
              {/each}
            </Select>
``

**Comment by cananda** [comment]
bind:selected is not working for me either (nor bind:selected={<name>} for one select or more than one select input in the same form. Is there any working example I could look at?

When I use bind:selected on both select inputs on the same form, I get "Attributes need to be unique"

**Comment by metonym** [comment]
Can you please provide a repro?

The following code demonstrates the following:

- initial `selected` value
- updating the `Select` should update the bound `selected` value
- programmatically updating `selected` should update the value

https://svelte.dev/repl/f1d79bc28ad34179b7ce556d85948c11?version=3.48.0

```svelte
<script>
  import { Select, SelectItem } from "carbon-components-svelte";

  let values = ["white", "g10", "g90", "g100"];
  let selected = "g10";
</script>

<Select labelText="Carbon theme" bind:selected>
  {#each values as value}
    <SelectItem {value} text={value} />
  {/each}
</Select>

<button
  on:click={() => {
    selected = "g90";
  }}
>
  Update to "g90"
</button>

{selected}

```

**Comment by cananda** [comment]
@metonym my apologies, this was a bug in my code. I should have provided an example.
I had bind:selected and selected = {} in my component which caused duplicate issue.  I originally thought this was an issue because of multiple selects in the same form was causing an issue.  This is working as expected. Thank you for the detailed response and the code snippet which helped identify the issue.

**Comment by nicrizzo-guess** [comment]
@metonym maybe I'm wrong, but maybe selected should match the selected value if unassigned, like the native select: https://svelte.dev/tutorial/select-bindings
Now its starting value is undefined

**Comment by metonym** [comment]
@nicrizzo-guess In the example, `selected` is set to the first value if `undefined`. Is that the expected behavior that you propose?

In the REPL, logging the value of `selected` shows the following:

```
undefined
{ id: 1 ,text: "Where did you go to school?" }
```

**Comment by nicrizzo-guess** [comment]
Yes, because otherwise I have to use a ref to the native select (__typesSelect in my case) and assign its value to the variable:
```
let newProjectType;
...
$:{
  if(__typesSelect){
    newProjectType = __typesSelect.value;
  }
}
```
Does it sound right to you?
Thank you very much!

**Comment by metonym** [comment]
Fixed in [v0.66.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.66.0).

If the `selected` prop is `undefined` or not provided to `Select`, its value will be set to the value of the first `SelectItem` to emulate the native `select` behavior.
